### ISSUE TITLE
[Crash report] - No such file or directory Proton-6.12-GE-1.tar.gz

### ISSUE THREAD

**Issue by Brolsorro** [bottlesdevs/Bottles#392]
This crash report was generated by Bottles.

**Details**
* **package**: Other

**Log**
```python3
[Errno 2] Нет такого файла или каталога: '/home/goto/.local/share/bottles/temp/Proton-6.12-GE-1.tar.gz'  File "/usr/share/bottles/bottles/utils.py", line 198, in target
    result = self.task_func(*args, **kwargs)

  File "/usr/share/bottles/bottles/runner.py", line 234, in async_install_component
    download = self.download_component(component_type,

  File "/usr/share/bottles/bottles/runner.py", line 218, in download_component
    os.remove(file_path)

```

**Comment by mirkobrombin** [comment]
I need more info:
- distribution name
- package type
- package version

**Comment by Brolsorro** [comment]
Ubuntu 21.04
Deb package
Version: 2021.7.28-treviso

The bug appeared after downloading Proton 6.12 GE-2 and interrupting this process by closing the window with versions of Wine and exiting to the main menu, where the list of bottles.

**Comment by mirkobrombin** [comment]
Oh ok this is the unstable build. Can you please try the AppImage?
I need to understand if it is a problem related to the deb package or all.

**Comment by Brolsorro** [comment]
The Appimage with version 2021.7.14-treviso also has this bug!

**Comment by Brolsorro** [comment]
If the bug does not reproduce: close the application,after the above steps, and reopen

**Comment by mirkobrombin** [comment]
I've tried the following methods:

**Starting the runner installation and closing the preferences window**
The installation simply continued in the background.

**Killing the process during the installation and trying to start it again**
Bottles found and deleted the partial download in the temp path and I was able to restart the download.

I can't replicate your bug, maybe I forget some steps?

**Comment by Brolsorro** [comment]
Played every other time. Strange, but this bug has stopped playing for me
The bug was reproduced using these steps
**Steps**
1. Start Bottles, check any created bottle
2. Select any bottle and enter its settings
3. Go to "Preferences" - "Runner" - "Settings" - "Runners"
4. Click for download, e.g. Proton 6.8 GE-2 and abort download by closing application
5. Repeat steps 1 to 3
6. Click for download Proton 6.8 GE-2 and the first time the download button may not work
7. Wait for **full download**
8. Close an application
9. Run the application and this bug may appear

**Comment by mirkobrombin** [comment]
> Played every other time. Strange, but this bug has stopped playing for me
> The bug was reproduced using these steps
> **Steps**
> 
> 1. Start Bottles, check any created bottle
> 2. Select any bottle and enter its settings
> 3. Go to "Preferences" - "Runner" - "Settings" - "Runners"
> 4. Click for download, e.g. Proton 6.8 GE-2 and abort download by closing application
> 5. Repeat steps 1 to 3
> 6. Click for download Proton 6.8 GE-2 and the first time the download button may not work
> 7. Wait for **full download**
> 8. Close an application
> 9. Run the application and this bug may appear

Component installation performs two tasks: download and extract. Perhaps the problem is caused by the second task, perhaps at the end of the download I notify the end of the installation even if the extraction is still in progress. So closing the application while extracting, results in bugs.

**Comment by mirkobrombin** [comment]
Now Bottles handle this exception. Should be fixed in 2021.7.28
